samcli/lib/package/packageable_resources.py

```diff
@@ -92,7 +92,7 @@ class ResourceZip(Resource):
     """
 
     RESOURCE_TYPE: Optional[str] = None
-    PROPERTY_NAME: Optional[str] = None
+    PROPERTY_NAME: str = ""
```

Comment:
I cross-checked the code, it is a safe change as all subclasses override this attribute.